### PR TITLE
Debugging info in CompilerContext.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/codegen/CompilerContext.cpp [hunk 1]
```diff
@@ -32,6 +32,7 @@
 #include <libsolidity/inlineasm/AsmCodeGen.h>
 #include <libsolidity/inlineasm/AsmAnalysis.h>
 #include <libsolidity/inlineasm/AsmAnalysisInfo.h>
+#include <libsolidity/inlineasm/AsmPrinter.h>
```
Comment (axic) [inline libsolidity/codegen/CompilerContext.cpp]: I think this is not needed?

File: libsolidity/codegen/CompilerContext.cpp [hunk 2]
```diff
@@ -313,10 +314,15 @@ void CompilerContext::appendInlineAssembly(
 	ErrorReporter errorReporter(errors);
 	auto scanner = make_shared<Scanner>(CharStream(_assembly), "--CODEGEN--");
 	auto parserResult = assembly::Parser(errorReporter).parse(scanner);
-	if (!parserResult || !errorReporter.errors().empty())
+//	cout << assembly::AsmPrinter()(*parserResult) << endl;
```
Comment (axic) [inline libsolidity/codegen/CompilerContext.cpp]: Stray line.

File: libsolidity/codegen/CompilerContext.cpp [hunk 3]
```diff
@@ -313,10 +314,15 @@ void CompilerContext::appendInlineAssembly(
...
+	assembly::AsmAnalysisInfo analysisInfo;
+	bool analyzerResult = false;
+	if (parserResult)
+		analyzerResult = assembly::AsmAnalyzer(analysisInfo, errorReporter, false, identifierAccess.resolve).analyze(*parserResult);
+	if (!parserResult || !errorReporter.errors().empty() || !analyzerResult)
```
Comment (axic) [inline libsolidity/codegen/CompilerContext.cpp:330]: Could be reduced to `!analyzerResult || !errorReporter..`

File: libsolidity/codegen/CompilerContext.cpp (hunk 2 above)
Comment (chriseth) [reply libsolidity/codegen/CompilerContext.cpp]: That's one of the two main changes of this PR. I would like to have a way to output the intermediate code, even if analysis is successful.

Comment (axic) [reply libsolidity/codegen/CompilerContext.cpp]: Hm? It is a comment in this PR?

Comment (chriseth) [reply libsolidity/codegen/CompilerContext.cpp]: Yes and you can temporarily uncomment it if you need the info.

Comment (axic) [reply libsolidity/codegen/CompilerContext.cpp]: Why not put it in an ifdef then?

Comment (chriseth) [reply libsolidity/codegen/CompilerContext.cpp]: I can do that.

**Review by axic — APPROVED** [review]

**Pull request merged**